### PR TITLE
Bump flake8 range to include 5.0 and refresh deps

### PR DESCRIPTION
**Pull request by Secrus**
Resolves: #98 

I took it upon myself to update `pyproject.toml` file: 
- the removed classifiers are added by Poetry on build, based on supported version ranges (old ones were still pointing to Python 3.5 as supported)
- bumped project version
- bumped Python to 3.7 (new `flake8` requires 3.6.1+, but as 3.6 is no longer supported, I guess this could be as good time as any)
- bumped dev dependencies to the newest versions (btw `twine` is not really needed, since Poetry can handle uploads to PyPI)
- added a missing `poetry.lock` file (required for consistent builds)

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -10,12 +10,6 @@ repository = "https://github.com/rocioar/flake8-django"
 classifiers=[
     "Framework :: Flake8",
     "License :: OSI Approved :: MIT License",
-    "Programming Language :: Python",
-    "Programming Language :: Python :: 3",
-    "Programming Language :: Python :: 3.5",
-    "Programming Language :: Python :: 3.6",
-    "Programming Language :: Python :: 3.7",
-    "Programming Language :: Python :: 3.8",
```
Comment (rocioar) [inline pyproject.toml:18]: Do we still need to have the label for 3.8 and 3.7 since they will be supported?

Comment (Secrus) [reply pyproject.toml:18]: Poetry automatically adds whole range of supported Python version classifiers.  See [`cleo` pyproject](https://github.com/python-poetry/cleo/blob/1.0.0a5/pyproject.toml) and [`cleo` on PyPI](https://pypi.org/project/cleo/1.0.0a5/).

Comment (rocioar) [reply pyproject.toml:18]: nice!

File: pyproject.toml [hunk 2]
```diff
@@ -29,10 +23,10 @@ requires = ["poetry_core>=1.0.0"]
 build-backend = "poetry.core.masonry.api"
 
 [tool.poetry.dependencies]
-python = "^3.6"
-flake8 = ">=3.8.4,<5"
+python = "^3.7"
+flake8 = ">=3.8.4,<6"
```
Comment (rocioar) [inline pyproject.toml:27]: Can you add a note about this bump to the CHANGELOG? I'll merge this and publish the version after that's updated. Thank you!

Comment (Secrus) [reply pyproject.toml:27]: Done.

**Pull request merged**